core/src/main/java/uk/gov/justice/services/core/annotation/Component.java

```diff
@@ -1,83 +1,74 @@
 package uk.gov.justice.services.core.annotation;
 
 import static java.lang.String.format;
-import static java.util.Arrays.stream;
-import static java.util.stream.Collectors.joining;
 
-import java.util.stream.Stream;
-
-import javax.jms.Destination;
-import javax.jms.Queue;
-import javax.jms.Topic;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
- * Enum representing all the service components.
+ * Class representing all the service components.
  */
-public enum Component {
+public final class Component {
+
+    public static final String COMMAND_API = "COMMAND_API";
+    public static final String COMMAND_CONTROLLER = "COMMAND_CONTROLLER";
+    public static final String COMMAND_HANDLER = "COMMAND_HANDLER";
 
-    COMMAND_API("command", "api", Queue.class),
-    COMMAND_CONTROLLER("command", "controller", Queue.class),
-    COMMAND_HANDLER("command", "handler", Queue.class),
-    EVENT_LISTENER("event", "listener", Topic.class),
-    EVENT_PROCESSOR("event", "processor", Topic.class),
-    EVENT_API("event", "api", null),
-    QUERY_API("query", "api", null),
-    QUERY_CONTROLLER("query", "controller", null),
-    QUERY_VIEW("query", "view", null);
+    public static final String EVENT_API = "EVENT_API";
+    public static final String EVENT_PROCESSOR = "EVENT_PROCESSOR";
+    public static final String EVENT_LISTENER = "EVENT_LISTENER";
 
-    private final String pillar;
-    private final String tier;
-    private final Class<? extends Destination> inputDestinationType;
+    public static final String QUERY_API = "QUERY_API";
+    public static final String QUERY_CONTROLLER = "QUERY_CONTROLLER";
+    public static final String QUERY_VIEW = "QUERY_VIEW";
 
-    Component(final String pillar, final String tier, final Class<? extends Destination> inputDestinationType) {
-        this.pillar = pillar;
-        this.tier = tier;
-        this.inputDestinationType = inputDestinationType;
+    private Component() {
     }
 
+    private static final List<String> components = new ArrayList<String>() {
+
+        private static final long serialVersionUID = -7403626856706763685L;
```

Comment:
Is this in the correct position? or is it just github making it look a bit weird.